### ISSUE TITLE
Openshift.clean() doesn't take KEEP_LABEL into account, regression 0.12 -> 0.13

### ISSUE THREAD

**Issue by pkremens** [xtf-cz/xtf#306]
```java
public class CleanupTest {
            List<ConfigMap> configMapList = Stream.of(
                            new ConfigMapBuilder("keep-label-map").addLabel(OpenShift.KEEP_LABEL, null).build(),
                            new ConfigMapBuilder("null-label-map").build(),
                            new ConfigMapBuilder("arbitrary-label-map").build()
            ).collect(Collectors.toList());
     
            private void createConfigMaps() {
                    for (ConfigMap configMap : configMapList) {
                            OpenShifts.master().createConfigMap(configMap);
                    }
                    new SimpleWaiter(() -> OpenShifts.master().getConfigMaps().size() == 3).waitFor();
            }
     
            private void deleteConfigMaps() {
                    for (ConfigMap configMap : OpenShifts.master().getConfigMaps()) {
                            OpenShifts.master().deleteConfigMap(configMap);
                    }
                    new SimpleWaiter(() -> OpenShifts.master().getConfigMaps().size() == 0).waitFor();
            }
     
            private void deleteConfigMaps(List<ConfigMap> removables) {
                    int expected = OpenShifts.master().getConfigMaps().size() - removables.size();
                    for (ConfigMap configMap : removables) {
                            OpenShifts.master().deleteConfigMap(configMap);
                    }
                    new SimpleWaiter(() -> OpenShifts.master().getConfigMaps().size() == expected).waitFor();
            }
     
            @BeforeEach
            public void setupEnv() {
                    deleteConfigMaps();
                    createConfigMaps();
            }
     
            @Test
            public void openshiftCleanTest() {
                    OpenShifts.master().clean();
                    Assertions.assertEquals(1, OpenShifts.master().getConfigMaps().size());
            }
     
            @Test
            public void customCleanTest() {
                    List<ConfigMap> removables = OpenShifts.master().getConfigMaps().stream()
                                    .filter(withoutKeepLabel())
                                    .collect(Collectors.toList());
                    deleteConfigMaps(removables);
                    Assertions.assertEquals(1, OpenShifts.master().getConfigMaps().size());
            }
     
            private Predicate<HasMetadata> withoutKeepLabel() {
                    return (hasMetadata) -> {
                            if (hasMetadata.getMetadata().getLabels() != null) {
                                    return !(hasMetadata.getMetadata().getLabels().containsKey(OpenShift.KEEP_LABEL));
                            } else {
                                    return true;
                            }
                    };
            }
    }
```
**0.12**
```sh
mvn clean test -Dtest=CleanupTest -Dxtf.version=0.12

[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
```

**0.13**
```sh
mvn clean test -Dtest=CleanupTest -Dxtf.version=0.13

[ERROR] Failures: 
[ERROR]   CleanupTest.openshiftCleanTest:59 expected: <1> but was: <0>    
    org.opentest4j.AssertionFailedError:
    Expected :1
    Actual   :0
            at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
            at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
            at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
            at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
            at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:510)
            at com.redhat.xpaas.eap.operator.junit.listener.CleanupTest.openshiftCleanTest(CleanupTest.java:59)
```

**Comment by pkremens** [comment]
**0.12**
```java
configMaps().withoutLabel(KEEP_LABEL).delete();
```
**0.13**
```java
removables.addAll(configMaps().withoutLabel(OpenShift.KEEP_LABEL).list().getItems());
```
There doesn't seem to be much of a difference, maybe the issue itself is hidden in kubernetes client.

**Comment by pkremens** [comment]
Indeed, it seems like the problem is located in one of the dependencies:
```java
	@Test
	public void withoutLabelTest() {
		System.out.println(openshift.configMaps().withoutLabel(KEEP_LABEL).list().getItems().size());
	}
```
**0.12**
```
 mvn clean test -Dtest=CleanupTest#withoutLabelTest -Dxtf.version=0.12
2
```
**0.13**
```
 mvn clean test -Dtest=CleanupTest#withoutLabelTest -Dxtf.version=0.13
3
```

**Comment by pkremens** [comment]
with xtf 0.13
```java
	List<ConfigMap> configMapList = Stream.of(
			new ConfigMapBuilder("keep-label-map").addLabel(KEEP_LABEL, null).build(),
			new ConfigMapBuilder("null-label-map").build(),
			new ConfigMapBuilder("arbitrary-label-map").build()
	).collect(Collectors.toList());

openshift.configMaps().withoutLabel(KEEP_LABEL).list().getItems().size() == 3

openshift.configMaps().withLabel(KEEP_LABEL).list().getItems().size() == 1

openshift.configMaps().withLabelNotIn(KEEP_LABEL).list().getItems().size() = 2
```
`withLabelNotIn` seems to be the way to go here.



```

**Comment by pkremens** [comment]
PR sent: https://github.com/xtf-cz/xtf/pull/309

**Comment by Crumby** [comment]
Fixed by #309

**Comment by mchoma** [comment]
Would be nice to create also openshift client issue. 

Just note, With current implementation I have noticed `xtf.cz/keep=true` did not work for me. I have to use `xtf.cz/keep=`. But maybe it is how `withLabelNotIn(String var1, String... var2)` should work.
